docs/interchain-staking.md

```diff
@@ -0,0 +1,178 @@
+
+Either way comes with the question of whether it should be the parent chain that is verifying the evidence of slashable events, or simply trusting the child chain's commands to slash validators at the level of the parent. While there may be the ability to both, trusting the child chain makes the job of the parent chain much easier. In that scenario the parent chain doesn't need to have knowledge of the logic contained within the child chain that may be necessary for determining whether a slashable offense took place. Trusting the child chain to enforce its own rules puts a lot of trust in the child chain to not abuse the position and submit a wave of slashable commands that may reduce a validator to nothing. However this is ultimately the responsibility of the validator to ensure that the slashable risk of validating on a child chain is worth the expected rewards. This includes determining whether the state machine of the child chain includes logic that the validator is comfortable with.
+
+Limits on a child chain's ability to slash a parent chain validator may also be imposed at the parent chain level. For instance it is possible to make it a required parameter of a child chain within the Chain Registry to provide a maximum slashing rate. This would ensure that a validator knows even if they violate all the rules of a child chain, they can still limit the damage to their stake by some amount. It is also possible to enforce a parameter at the level of the parent chain that prevents a validator from joining any combination of child chain validator sets that results in a combined slashable rate over some threshold. For instance if there are 3 child chains that each have maximum slashable rates of 33% over X blocks, and the parent chain has a safety limit for validators that require them to stay below a total of 90% slashable events over X blocks, no validator would be allowed to join the Chain Registry for all 3 of those child chains. This parameter on the parent chain should be a governance parameter that can be adjusted by token staked voting to reflect the risk appetite of the actual validator set of the parent chain.
```

Comment:
Mitigation helps, though its important to note that it's not just the validator violating rules. A 2/3 cabal of validators on child chain can arbitrarily slash any other child chains validator's parent stake up to the maximum slashable amount

Reply:
very interesting....
this attack would be one child chain targeting another (possibly competitive) child chain by punishing its own shared validators? pretty nasty behaviour, i am inclined to ask why that validator got involved in such sketchy chains in the first place but I guess it could be a lot less about the chain than the other validators...

this would be more likely on child chains that have low cost of corruption due to low participation from parent validator sets? Even one or a small number of large validators could join the child chain just to get rid of general validator competition.

this makes me more inclined to prioritize the V0 version of full validator sets only...